src/components/MarketMySharesForm/index.js

```diff
@@ -143,58 +147,42 @@ class MarketMySharesForm extends Component {
 
     const resolvedOrClosed = isMarketClosed(market) || isMarketResolved(market)
 
-    Object.keys(marketShares).forEach((shareKey) => {
-      const share = marketShares[shareKey]
-      const probability = calcLMSRMarginalPrice({
-        netOutcomeTokensSold: market.netOutcomeTokensSold.slice(0),
-        funding: market.funding,
-        outcomeTokenIndex: share.outcomeToken.index,
-      })
-      const maximumWin = calcLMSROutcomeTokenCount({
-        netOutcomeTokensSold: market.netOutcomeTokensSold.slice(0),
-        funding: market.funding,
-        outcomeTokenIndex: share.outcomeToken.index,
-        cost: share.balance,
-      })
+    Object.keys(marketShares).forEach((shareId) => {
+      const share = marketShares[shareId]
+      const colorScheme = share.event.type === OUTCOME_TYPES.SCALAR ? COLOR_SCHEME_SCALAR : COLOR_SCHEME_DEFAULT
+      const outcomeColorStyle = { backgroundColor: colorScheme[share.outcomeToken.index] }
 
       tableRows.push(<tr className="marketMyShares__share" key={share.id}>
         <td>
-          <div
-            className="shareOutcome__color"
-            style={{ backgroundColor: COLOR_SCHEME_DEFAULT[share.outcomeToken.index] }}
-          />
+          <div className="shareOutcome__color" style={outcomeColorStyle} />
         </td>
         <td className="">{getOutcomeName(market, share.outcomeToken.index)}</td>
         <td>
           {Decimal(share.balance)
-            .div(1e18)
-            .gte(LOWEST_DISPLAYED_VALUE) ? (
-              <DecimalValue value={Decimal(share.balance).div(1e18)} />
+              .div(1e18)
+              .gte(LOWEST_DISPLAYED_VALUE) ? (
+                <DecimalValue value={Decimal(share.balance).div(1e18)} />
             ) : (
               `< ${LOWEST_DISPLAYED_VALUE}`
             )}
         </td>
-        <td>
-          <DecimalValue value={maximumWin.mul(probability).div(1e18)} />&nbsp;
-          <CurrencyName collateralToken={market.event.collateralToken} />
-        </td>
         <td>
           {!resolvedOrClosed && (
-            <a
-              href="javascript:void(0);"
-              className="marketMyShares__sellButton"
-              onClick={e => this.handleShowSellView(e, share.id)}
-            >
+          <a
+            href="javascript:void(0);"
+            className="marketMyShares__sellButton"
+            onClick={e => this.handleShowSellView(e, share.id)}
```

Comment:
Arrow functions should be avoided in the render function, otherwise, you will re-render every time a prop or states or parent changes.

Also, try to change the ```<a>``` and use ```Link```, it should work I think.

Reply:
I know we need to avoid this, but I think the reason is a different one. A prop or a state change will (and should) cause re-renders, but we're leaking memory by _recreating_ the function every render. I think also the performance impact on this is very very low.

Take a look at this article from the React Docs: https://reactjs.org/docs/faq-functions.html#is-it-ok-to-use-arrow-functions-in-render-methods